src/PIL/GimpPaletteFile.py

```diff
@@ -26,30 +26,28 @@ class GimpPaletteFile:
 
     def __init__(self, fp):
 
-        self.palette = [o8(i) * 3 for i in range(256)]
+        palette = bytearray(b"".join([o8(i) * 3 for i in range(256)]))
 
         if fp.readline()[:12] != b"GIMP Palette":
             raise SyntaxError("not a GIMP palette file")
 
-        for i in range(256):
-
-            s = fp.readline()
-            if not s:
-                break
-
+        index = 0
+        for s in fp:
             # skip fields and comment lines
             if re.match(rb"\w+:|#", s):
                 continue
             if len(s) > 100:
                 raise SyntaxError("bad palette file")
 
             v = tuple(map(int, s.split()[:3]))
-            if len(v) != 3:
```

Comment:
This change seems unnecessary? The previous line has already limited the number of items to 3.